### PR TITLE
[bunyan] remove '@theia/bunyan' extension

### LINKED ISSUE #6637: [question] why is '@theia/bunyan' part of the main repository
**Description**

I have a question regarding the `@theia/bunyan` extension present in the main repository.
It looks like there are no extensions (part of the main repo) that depend or use the extension, and it is not included in the `example-browser` or `example-electron` applications. It also does not seem to be updated frequently. Is it a source for clean-up, or is there a reason it is present in the main repo?

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #6637

_Motivation:_
- the extension is not used by any other extension present in the main repo.
- the extension is not part of the `example-browser` or `example-electron` application, therefore it cannot be tested properly.
- since it is not part of any other extension, or tested, it is likely to be un-maintained.
- removing the extension also has a positive side-effect on build and CI time

#### How to test
- the PR was tested by verifying if any `@theia/bunyan` dependencies were present, building the project and ensuring that tests successfully passed when removed.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
I also noticed the dependency `bunyan` and `@types/bunyan` are declared but unused at the moment, should I clean those up as well?

https://github.com/eclipse-theia/theia/blob/87df21b291883ad46dc00d53131fab109a6fbda5/dev-packages/application-manager/package.json#L39

https://github.com/eclipse-theia/theia/blob/87df21b291883ad46dc00d53131fab109a6fbda5/packages/core/package.json#L13

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Breaking changes:
   Most browsers define a page as served from secure context if its url has `https` scheme. For local testing `localhost` is treated as a secure context as well.
   Unfortunately, it does not work nicely in FireFox, since it does not treat subdomains of localhost as secure as well, compare to Chrome.
   If you want to test with FireFox you can configure it as described [here](https://github.com/eclipse-theia/theia/pull/6465#issuecomment-556443218).
+  - [bunyan] removed `@theia/bunyan` extension []()
```
Comment (marcdumais-work) [inline CHANGELOG.md]: We could add the following link, that shows the last published version of this extension's source code: https://github.com/eclipse-theia/theia/tree/b92a5673de1e9d1bdc85e6200486b92394200579/packages/bunyan

Comment (vince-fugnitto) [reply CHANGELOG.md]: Updated based on your suggestion :)

**Comment by vince-fugnitto** [comment]
@marcdumais-work I added an extra commit which takes care of removing the `bunyan` and `@types/bunyan` dependencies from `application-manager` and `core` respectfully.

**Comment by marcdumais-work** [comment]
Looking good - with the latest version, the only `bunyan` reference I find when grepping the repo is from `CHANGELOG`: 

gitpod /workspace/theia $ grep -R -i bunyan *
CHANGELOG.md:  - [bunyan] removed [`@theia/bunyan`](https://github.com/eclipse-theia/theia/tree/b92a5673de1e9d1bdc85e6200486b92394200579/packages/bunyan) extension [#6651](https://github.com/eclipse-theia/theia/pull/6651)

**Comment by vince-fugnitto** [comment]
> LGTM, but we should probably wait at least a day before merging, to give community members time to potentially object.

Sounds good! :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ Breaking changes:
   Most browsers define a page as served from secure context if its url has `https` scheme. For local testing `localhost` is treated as a secure context as well.
   Unfortunately, it does not work nicely in FireFox, since it does not treat subdomains of localhost as secure as well, compare to Chrome.
   If you want to test with FireFox you can configure it as described [here](https://github.com/eclipse-theia/theia/pull/6465#issuecomment-556443218).
+  - [bunyan] removed [`@theia/bunyan`](https://github.com/eclipse-theia/theia/tree/b92a5673de1e9d1bdc85e6200486b92394200579/packages/bunyan) extension [#6651](https://github.com/eclipse-theia/theia/pull/6651)
```
Comment (akosyakov) [inline CHANGELOG.md]: wrong alignment? Should be aligned with L14?

**Comment by akosyakov** [comment]
> LGTM, but we should probably wait at least a day before merging, to give community members time to potentially object.

Maybe announce on Spectrum.

File: CHANGELOG.md (hunk 2 above)
Comment (vince-fugnitto) [reply CHANGELOG.md]: Updated! Thank you! :)

**Comment by vince-fugnitto** [comment]
> > LGTM, but we should probably wait at least a day before merging, to give community members time to potentially object.
> 
> Maybe announce on Spectrum, just to make sure. But we can always reincarnate it as a separate repo if there anyone eager to maintain it.

I created a post on Spectrum as well so its more visible to the community: https://spectrum.chat/theia/general/the-theia-bunyan-extension-to-be-removed~811cbba0-6977-449b-b7b0-ac5fb25b9231

**Comment by vince-fugnitto** [comment]
I'll give others a chance to view the PR if they'd like and merge on Monday.

**Pull request merged**